ballerina/fragment_visitor.bal

```diff
@@ -87,15 +84,26 @@ class FragmentVisitor {
         }
     }
 
-    isolated function getFragment(parser:Selection selection) returns parser:FragmentNode? {
-        parser:DocumentNode documentNode = <parser:DocumentNode>self.documentNode;
-        parser:FragmentNode? fragmentNode = documentNode.getFragment(selection.name);
-        if (fragmentNode is ()) {
-            string message = string`Unknown fragment "${selection.name}".`;
-            ErrorDetail errorDetail = getErrorDetailRecord(message, selection.location);
+    isolated function appendNamedFragmentFields(parser:FragmentNode fragmentNode) {
+        parser:DocumentNode documentNode = self.documentNode;
+        parser:FragmentNode? actualFragmentNode = documentNode.getFragment(fragmentNode.getName());
+        if actualFragmentNode is () {
+            string message = string`Unknown fragment "${fragmentNode.getName()}".`;
+            ErrorDetail errorDetail = getErrorDetailRecord(message, fragmentNode.getLocation());
             self.errors.push(errorDetail);
+        } else {
+            if !fragmentNode.isInlineFragment() && fragmentNode.getSelections().length() == 0 {
```

Comment:
What's the reason for using this condition `fragmentNode.getSelections().length() == 0` ?

Reply:
```graphql
query A($profileId:Int!) {
    ...data 
}

query B($profileId:Int!) {
    ...data
}

fragment data on Query {
    profile(id:$profileId) {
        ...people
    }
}

fragment people on Person {
    address {
        city
    }
}
```
when visiting a document like this, `selection[]` of  `...people` node in fragment `data` is also getting filled while procesing the `...data` node `selection[]` in query `A`. here,  `fragmentNode.getSelections().length() == 0` condition is used to avoid re-filling the `selection[]` in `...people` node when processing the `...data` node in query `B`.